src/server/scripts/Events/darkmoon_faire.cpp

```diff
@@ -0,0 +1,190 @@
+{
+    EVENT_SELINA_TALK_COOLDOWN              = 1
+};
+
+enum Gossips
+{
+    GOSSIP_MENU_SELINA_POIS                 = 13076,
+    GOSSIP_MENU_SELINA_ITEM                 = 13113
+};
+
+enum GossipMenuOptions
+{
+    GOSSIP_MENU_OPTION_TONK_ARENA_POI       = 0,
+    GOSSIP_MENU_OPTION_CANNON_POI           = 1,
+    GOSSIP_MENU_OPTION_WHACK_A_GNOLL_POI    = 2,
+    GOSSIP_MENU_OPTION_RING_TOSS_POI        = 3,
+    GOSSIP_MENU_OPTION_SHOOTING_GALLERY_POI = 4,
+    GOSSIP_MENU_OPTION_FORTUNE_TELLER_POI   = 5
+};
+
+enum Miscellanea
+{
+    DATA_ID_SELINA_TALK                     = 1,
+    DATA_VALUE_SELINA_TEXT_WELCOME          = 1
+};
+
+enum PointsOfInterest
+{
+    POI_WHACK_A_GNOLL                       = 2716,
+    POI_CANNON                              = 2717,
+    POI_SHOOTING_GALLERY                    = 2718,
+    POI_TONK_ARENA                          = 2719,
+    POI_FORTUNE_TELLER                      = 2720,
+    POI_RING_TOSS                           = 2721
+};
+
+enum Spells
+{
+    SPELL_REPLACE_DARKMOON_ADVENTURES_GUIDE = 103413
+};
+
+enum Texts
+{
+    SAY_WELCOME                             = 0
+};
+
+// 10445 - Selina Dourman
+struct npc_selina_dourman : public ScriptedAI
+{
+    npc_selina_dourman(Creature* creature) : ScriptedAI(creature) { }
+
+    bool OnGossipSelect(Player* player, uint32 menuId, uint32 gossipListId) override
+    {
+        switch (menuId)
+        {
+            case GOSSIP_MENU_SELINA_POIS:
+            {
+                uint32 poiId = 0;
+                switch (gossipListId)
+                {
+                    case GOSSIP_MENU_OPTION_TONK_ARENA_POI:
+                        poiId = POI_TONK_ARENA;
+                        break;
+                    case GOSSIP_MENU_OPTION_CANNON_POI:
+                        poiId = POI_CANNON;
+                        break;
+                    case GOSSIP_MENU_OPTION_WHACK_A_GNOLL_POI:
+                        poiId = POI_WHACK_A_GNOLL;
+                        break;
+                    case GOSSIP_MENU_OPTION_RING_TOSS_POI:
+                        poiId = POI_RING_TOSS;
+                        break;
+                    case GOSSIP_MENU_OPTION_SHOOTING_GALLERY_POI:
+                        poiId = POI_SHOOTING_GALLERY;
+                        break;
+                    case GOSSIP_MENU_OPTION_FORTUNE_TELLER_POI:
+                        poiId = POI_FORTUNE_TELLER;
+                        break;
+                    default:
+                        break; 
+                }
+                if (poiId)
+                    player->PlayerTalkClass->SendPointOfInterest(poiId);
+                break;
+            }
+            case GOSSIP_MENU_SELINA_ITEM:
+                me->CastSpell(player, SPELL_REPLACE_DARKMOON_ADVENTURES_GUIDE);
+                CloseGossipMenuFor(player);
+                break;
+            default:
+                break;
+        }
+
+        return false;
+    }
+
+    void SetData(uint32 id, uint32 value) override
```

Comment:
you can move the entire code of theis function into the method which is being called by the areatrigger (in this case `SetTalkTarget`. You might want to rename it to something like `DoWelcomeTalk` or something when you've moved it there.